datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -1036,12 +1193,50 @@ async fn collect_left_input(
     let data = JoinLeftData::new(
         hashmap,
         single_batch,
-        left_values,
+        left_values.clone(),
         Mutex::new(visited_indices_bitmap),
         AtomicUsize::new(probe_threads_count),
         reservation,
     );
 
+    // Update dynamic filter with min/max bounds if provided
+    if num_rows > 0 {
+        let bounds = compute_bounds(&left_values)?;
```

Comment:
For joins with a very small number of rows perhaps could use the values themselves to create a filter based on equality (`a==2||a==10`) which makes it a bit more selective with filter pushdown.

Reply:
Yeah makes sense!